### DISCUSSION TITLE
Welcome to Flux Discussions!

### DISCUSSION THREAD

**Discussion by dvmarinoff** [dvmarinoff/Auuki#18] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by outdated-dev** [comment]
How do i install and run this? some steps pls..

And which trainers have u tested this with?

**Reply by dvmarinoff** [reply]
Hi, this App is still in early development, my plans are to release the first stable version as a web based demo until the end of Jan 2021. Desktop versions will follow later. The web based version will work directly in the browser, no installation needed.

**Platform Support:** 
Supported are Chrome, Edge, Opera and Samsung Internet on Android, Linux and Windows. Given the limitations of the web platform on iOS devices, the web based version won't be available there. 

**Trainer Support:**
Supported are all Trainers that implement the FTMS bluetooth profile, meaning most smart trainers released in the last 3 years should work with the stable version. I am developing and testing directly with Tacx trainers (Flux 1 / S / 2), and soon I'll add Elite trainers too.

**Demo:**
Meanwhile you can try the web demo at the github pages of the repo: [Demo](https://dvmarinoff.github.io/Flux/), open it with Google Chrome or Edge for best results.

Keep in mind I am actively developing now and using the app for my daily training sessions so things will constantly change or brake until release. Readme, Instructions, How To's and more are in the pipeline, just haven't gotten to them yet.

**Reply by outdated-dev** [reply]
I tested the web version with my tacx flux s trainer and it works perfectly. Awesome job with the app!

Unfortunately the import feature did not work & I see you have an open issue on the same. Let me know if there is a way I can help fix such issues.

**Reply by dvmarinoff** [reply]
Thanks!
I've fixed the import issue, but the application is currently not capable of remembering imported workouts.
I'll implement this part later with IndexDB.
As for collaboration I'll move the project to a pull request model once I got the essential features working
and make the code more friendly. Right now I am figuring a lot of stuff like ANT+ and Web Serial API,
and I'll have to re-think the whole connections UI to incorporate ANT+, so everything is changing a lot.
You are welcome to open an issue if you find a bug or have particular feature enhancement in mind.

**Reply by RonBrumfield** [reply]
Dimitar, great work.  I plan to test this on a _**Stages SB20**_  ([https://stagescycling.com/us/stagesbike/](url)) over the next few days. The SB20 Smart Bike is less than a year old and gaining popularity.  Thanks for pulling this together!  I love that you have added ZWO functionality, fantastic. 

If it works with the SB20, I have a few ideas that may add to the real-world usefulness for a lot of smart-trainer-bike owners.  

Best regards, Ron

**Reply by svart-ravn** [reply]
@dvmarinoff , @outdated-dev , how I can install it? I did not find any guide. thanks

**Reply by dvmarinoff** [reply]
Hi @svart-ravn

If you are asking about the installable web-app this feature is off for the moment you can track the progress on it here: https://github.com/dvmarinoff/Flux/issues/129

But you can use the regular web-app at: https://flux-three.vercel.app/

If you want to just run the project locally check here: https://github.com/dvmarinoff/Flux/discussions/127

**Comment by lawrencemaroun** [comment]
@dvmarinoff I discovered this repo today from a Reddit thread. Very excited to try the app. I was looking for info about how to make a smart trainer app because I wanted to make my own, but I think instead I will try to contribute to this if you like my ideas and welcome the contribution.

I have two features in mind that I would like to create

1) Procedurally generated workouts. User selects a time, workout type, and difficulty offset. Optional modifiers can be activated like adding sprints to the beginning of each set of intervals.

2) A training game called "Guess Watt". The app sets the trainer to a hidden power level, moves it up and down for a bit, and tells the rider to guess how many watts they are making. Then reveals the wattage and repeats. The goal is to improve one's ability to estimate their power output, helping with outdoor pacing for those who don't own a power meter.

Since the first one is not too UI heavy, I can start by writing a JS function to do the workout generation and see how it goes. If it seems like a good feature, we can talk about the UI side and how to implement it.

Let me know what you think

**Reply by dvmarinoff** [reply]
Hi @lawrencemaroun,

I like the ideas, especially number 2, it’s original and it could be fun.

About number 1 lets move the discussion to this issue: https://github.com/dvmarinoff/Flux/issues/31
And 2 here: https://github.com/dvmarinoff/Flux/issues/133

I am a bit busy with the iOS app right now, but will try to give you some guidance around the codebase.

**Reply by lawrencemaroun** [reply]
@dvmarinoff Sounds good! Glad you like it. I will focus on the workout algorithm for now, and hopefully you will have time later to provide guidance when it's time to integrate it. Will let you know when I have a working concept to explain my approach.

**Reply by Lunchtime0614** [reply]
For reference, I happen to have some .zwo files that I find myself loading to Flux to run.  So in the context of this idea of generating a workout, it would be nice to save it somewhere so you could come back to it - say like an integration to google drive dropbox...

**Reply by benjaminlin** [reply]
I installed google drive app on my computer so I can select the workout I stored on cloud.

**Reply by Lunchtime0614** [reply]
> I installed google drive app on my computer so I can select the workout I stored on cloud.

Yeah that works, I have quite a few of them so I have to do filtering from an excel file/database to find the one that I'm looking for.  I pretty much do the same as what your suggesting...I'm thinking though there could be tighter integration.
